plugins/router/f5/f5.go

```diff
@@ -0,0 +1,1743 @@
+// on the F5 BIG-IP host.  In particular, this method creates a traffic
+// policy and associates it with the vservers if necessary.
+func (f5 *f5LTM) Initialize() error {
+	err := f5.ensurePolicyExists(httpPolicyName)
+	if err != nil {
+		return err
+	}
+
+	if f5.httpVserver != "" {
+		err = f5.ensureVserverHasPolicy(f5.httpVserver, httpPolicyName)
+		if err != nil {
+			return err
+		}
+	}
+
+	if f5.httpsVserver != "" {
+		err = f5.ensurePolicyExists(httpsPolicyName)
+		if err != nil {
+			return err
+		}
+
+		err = f5.ensureVserverHasPolicy(f5.httpsVserver, httpsPolicyName)
+		if err != nil {
+			return err
+		}
+
+		if f5.httpVserver != "" {
+			err = f5.ensureVserverHasPolicy(f5.httpVserver, httpPolicyName)
+			if err != nil {
+				return err
+			}
+		}
+
+		err = f5.ensureDatagroupExists(passthroughRoutesDataGroupName)
+		if err != nil {
+			return err
+		}
+
+		err = f5.ensureDatagroupExists(passthroughHostsDataGroupName)
+		if err != nil {
+			return err
+		}
+
+		err = f5.ensureIRuleExists(sslPassthroughIRuleName, sslPassthroughIRule)
+		if err != nil {
+			return err
+		}
+
+		err = f5.ensureVserverHasIRule(f5.httpsVserver, sslPassthroughIRuleName)
+		if err != nil {
+			return err
+		}
+	}
+
+	glog.V(4).Infof("F5 initialization is complete.")
+
+	return nil
+}
+
+// Create a pool named poolname on F5.
+func (f5 *f5LTM) CreatePool(poolname string) error {
+	url := fmt.Sprintf("https://%s/mgmt/tm/ltm/pool", f5.host)
+
+	payload := struct {
+		Mode    string `json:"loadBalancingMode"`
+		Monitor string `json:"monitor"`
+		Name    string `json:"name"`
+	}{
+		Mode:    "round-robin",
+		Monitor: "/Common/http",
+		Name:    poolname,
+	}
+
+	err := f5.post(url, payload, nil)
+	if err != nil {
+		return err
+	}
+
+	// We don't really need to initialise f5.poolMembers[poolname] because
+	// we always check whether it is initialised before using it, but
+	// initialising it to an empty map here saves a REST call later the first
+	// time f5.PoolHasMember is invoked with poolname.
+	f5.poolMembers[poolname] = map[string]bool{}
+
+	glog.V(4).Infof("Pool %s created.", poolname)
+
+	return nil
+}
+
+// Delete the pool named poolname from F5, and delete
+// f5.poolMembers[poolname].
+func (f5 *f5LTM) DeletePool(poolname string) error {
+	url := fmt.Sprintf("https://%s/mgmt/tm/ltm/pool/%s", f5.host, poolname)
+
+	err := f5.delete(url, nil)
+	if err != nil {
+		return err
+	}
+
+	// Note: We *must* use delete here rather than merely assigning false because
+	// len() includes false items, and we want len() to return an accurate count
+	// of members.  Also, we probably save some memory by using delete.
+	delete(f5.poolMembers, poolname)
+
+	glog.V(4).Infof("Pool %s deleted.", poolname)
+
+	return nil
+}
+
+// Return f5.poolMembers[poolname], first initializing it from F5 if it is zero.
+func (f5 *f5LTM) GetPoolMembers(poolname string) (map[string]bool, error) {
+	members, ok := f5.poolMembers[poolname]
+	if ok {
+		return members, nil
+	}
```

Comment:
It could.  If that's a use-case we want to support, then we could add a check to invalidate the cache after some period of time has passed, or add a handler for `SIGUSR1` or similar to force a cache-dump (I like this option better), or even do both.